torch/csrc/jit/passes/onnx/eval_peephole.cpp

```diff
@@ -46,14 +46,18 @@ static void fuseConvBatchNorm(Block* b, ValueToParamPairMap& valsToParamsMap) {
       fuseConvBatchNorm(child_block, valsToParamsMap);
     }
     if (it->kind() == onnx::Conv) {
-      if (it->output()->uses().size() != 1) {
+      auto origconvNode = *it;
+      if (origconvNode->output()->uses().size() != 1)
         continue;
-      }
-      auto bnNode = it->output()->uses()[0].user;
-      if (bnNode->kind() != onnx::BatchNormalization) {
+
+      auto bnNode = origconvNode->output()->uses()[0].user;
+      if (bnNode->kind() != onnx::BatchNormalization)
         continue;
-      }
-      auto origconvNode = *it;
+      // Even when model is in eval mode, check if BatchNorm layer is in
+      // training mode
+      if (bnNode->outputs().size() != 1)
```

Comment:
does batchnorm has a training attribute to check with, or is this only achievable by looking at output count?

Reply:
Starting from opset 14, it has an attribute.